course.php

```diff
@@ -30,10 +32,27 @@
 $PAGE->set_heading($title);
 $PAGE->navbar->add($title);
 $PAGE->set_pagelayout('standard');
-$PAGE->blocks->add_region('content');
+
 // Check user is logged in and capable of grading.
 require_login($course, false);
 
+$block_manager = $PAGE->blocks;
+$block_manager->add_region('content');
+$block_manager->load_blocks(true);
+
+// Check if block instance does not yet exist
+$gs_exists = false;
+foreach ($block_manager->get_regions() as $region) {
+    foreach ($block_manager->get_blocks_for_region($region) as $block) {
+        if ($block->instance->blockname == "side_bar") {
```

Comment:
Should you not be checking for 'blocks-side_bar-course' ?

Reply:
What if a different sidebar block was present?  you cannot assume it will be the side bar course one right?

Reply:
No, as that would be more of a page visibility name as in applicable_formats(). The 'side_bar' here is the DOM block loaded in the region of the page.

Reply:
Well I expect it to be in that region - I am only limiting the scope of my check to the 'content' region. So if it exists elsewhere, then we continue to add it there.